### PR TITLE
Bugfix: Do not assume a fully functional matrix interface for mg matrices

### LINKED ISSUE #4613: More matrix-free / GMG  bugs from PR 4606
@tamiko it looks like there are more bugs introduced in https://github.com/dealii/dealii/pull/4606 related to geometric multigrid and matrix-free. Can you please run the corresponding test-suite on your branch to confirm? 

The errors are convoluted but roughly read

```
lac/la_parallel_vector.h:1496:9: error: request for member ‘initialize_dof_vector’ in ‘matrix’, which is of pointer type <....>
lac/la_parallel_vector.h:1507:9: error: request for member ‘initialize_dof_vector’ in ‘matrix’, which is of pointer type <.....>
```

### PR DESCRIPTION
**Pull request by tamiko**
Closes #4613

### CODE REVIEW

File: include/deal.II/multigrid/mg_block_smoother.h [hunk 1]
```diff
@@ -194,33 +194,12 @@ MGSmootherBlock<MatrixType, RelaxationType, number>::initialize (const MGMatrixT
...
-      {
-        s[i].vmult(v, u);
-      };
-      smoothers[i].Tvmult = [&s, i](BlockVector<number> &v,
-                                    const BlockVector<number> &u)
-      {
-        s[i].Tvmult(v, u);
-      };
+      // Workaround: Unfortunately, not every "m[i]" object has a
+      // rich enough interface to poppulate reinit_(domain|range)_vector.
```
Comment (kronbichler) [inline include/deal.II/multigrid/mg_block_smoother.h]: poppulate -> populate

File: include/deal.II/multigrid/mg_coarse.h [hunk 2]
```diff
@@ -376,8 +376,11 @@ ::MGCoarseGridLACIteration (SolverType               &s,
   :
   solver(&s, typeid(*this).name())
 {
-  matrix = linear_operator<VectorType>(m);
-  precondition = linear_operator<VectorType>(m, p);
+  // Workaround: Unfortunately, not every "m" object has a rich enough
+  // interface to poppulate reinit_(domain|range)_vector. Thus, supply an
```
Comment (kronbichler) [inline include/deal.II/multigrid/mg_coarse.h]: populate

File: include/deal.II/multigrid/mg_coarse.h [hunk 3]
```diff
@@ -398,8 +401,11 @@ ::initialize (SolverType               &s,
               const PreconditionerType &p)
 {
   solver = &s;
-  matrix = linear_operator<VectorType>(m);
-  precondition = linear_operator<VectorType>(m, p);
+  // Workaround: Unfortunately, not every "m" object has a rich enough
+  // interface to poppulate reinit_(domain|range)_vector. Thus, supply an
```
Comment (kronbichler) [inline include/deal.II/multigrid/mg_coarse.h]: populate

File: include/deal.II/multigrid/mg_coarse.h [hunk 4]
```diff
@@ -433,7 +439,10 @@ void
 MGCoarseGridLACIteration<SolverType, VectorType>
 ::set_matrix(const MatrixType &m)
 {
-  matrix = linear_operator<VectorType>(m);
+  // Workaround: Unfortunately, not every "m" object has a rich enough
+  // interface to poppulate reinit_(domain|range)_vector. Thus, supply an
```
Comment (kronbichler) [inline include/deal.II/multigrid/mg_coarse.h]: populate

File: include/deal.II/multigrid/mg_matrix.h [hunk 5]
```diff
@@ -185,20 +185,11 @@ namespace mg
...
-        {
-          p[level].vmult(v, u);
-        };
-        matrices[level].Tvmult = [&p, level](
-                                   VectorType &v, const VectorType &u)
-        {
-          p[level].Tvmult(v, u);
-        };
+        // Workaround: Unfortunately, not every "p[level]" object has a
+        // rich enough interface to poppulate reinit_(domain|range)_vector.
```
Comment (kronbichler) [inline include/deal.II/multigrid/mg_matrix.h]: populate

File: include/deal.II/multigrid/mg_smoother.h [hunk 6]
```diff
@@ -805,7 +805,11 @@ MGSmootherRelaxation<MatrixType, RelaxationType, VectorType>::initialize
 
   for (unsigned int i=min; i<=max; ++i)
     {
-      matrices[i] = linear_operator<VectorType>(m[i]);
+      // Workaround: Unfortunately, not every "m[i]" object has a rich
+      // enough interface to poppulate reinit_(domain|range)_vector. Thus,
```
Comment (kronbichler) [inline include/deal.II/multigrid/mg_smoother.h]: same here (and five more times in this file).

**Comment by tamiko** [comment]
@kronbichler Fixed.

**Pull request merged**